fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+                 "WHERE table_schema='{}' "
+                 "AND table_name='{}'").format(schema_name,
+                                               table_name)
+        result = self.execute_query_from_string(query)
+        colnames = []
+        for row in result:
+            colnames.append(row[0])
+        return colnames
+
+    def create_master_feature_view(self, feature_table_names):
+        """This generates a view of all the feature tables at
+        features.frontpage_features in the following form:
+
+        exampleid  | feature_1 | feature_2 ...    | feature_n
+        (integer)  | (bigint)  | (bigint)  ...    | (bigint)
+        --------------------------------------------------------
+        9          | 1         | 2         ...    | 34
+        10         | 2         | 2         ...    | 21
+        11         | 1         | 3         ...    | 36
+
+        A single row represents a single example along with the
+        features for that example. The purpose of the view is to
+        provide a simple way to select train and test sets.
+
+        Args:
+            feature_table_names [list of strings]: list of tables
+            that contain features that we would like to put into
+            the view
+        """
+
+        master_features = {}
+
+        for schema_and_table in feature_table_names:
+            schema_name = schema_and_table.split('.')[0]
+            table_name = schema_and_table.split('.')[1]
+            table_columns = self._get_column_tables_of_table(schema_name,
+                                                             table_name)
+            table_columns.remove("exampleid")
+            master_features.update({schema_and_table: table_columns})
+
+        columns_to_select = "foo.exampleid"
+        full_join_query = ("FROM ( (SELECT exampleid FROM      " 
+                           "features.undefended_frontpage_examples) foo ")
+
+        for table_num, table_name in enumerate(list(master_features.keys())):
+            prefix = 't{}.'.format(table_num)
+            prefixed_columns = [prefix + s for s in master_features[table_name]]
+            columns_to_select = columns_to_select + ', ' + ', '.join(prefixed_columns)
+            join_query = ("LEFT OUTER JOIN {name} t{num} "
+                          "ON foo.exampleid = t{num}.exampleid ").format(name=table_name,
+                                                                         num=table_num)
+            full_join_query = full_join_query + join_query
+
+        drop_view = "DROP VIEW IF EXISTS features.frontpage_features; "
+        self.execute_query_from_string(drop_view)
+
+        create_new_view = ("CREATE VIEW features.frontpage_features "
+                           "AS ( SELECT {} {} ));").format(columns_to_select,
+                                                          full_join_query)
+        self.execute_query_from_string(create_new_view)
+        return None
+
+
+def main():
```

Comment:
If we later create a script that runs everything from feature generation to model building for us (which we probably will) - then it is nice to have the feature generation steps defined in a `main()` function such that we have the option to have the master script runs it, or we run it directly as you suggest